src/Heatmap/HeatMapInterpolationPoints.swift

```diff
@@ -0,0 +1,399 @@
+/* Copyright (c) 2020 Google Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import Foundation
+import GooglePlaces
+import GoogleMaps
+import GoogleMapsUtils
+
+/// This class will create artificial points in surrounding locations with appropriate intensities interpolated by neighboring intensity values.
+class HeatMapInterpolationPoints {
+    
+    /// The input data set; each entry contains the coordinates and the 
+    private var data = [[Double]]()
+    
+    /// The heat map layer
+    private let heatMapLayer: GMUHeatmapTileLayer = GMUHeatmapTileLayer()
+    
+    /// The heat map colors and gradient
+    private var gradientColors = [UIColor.green, UIColor.red]
+    private var gradientStartPoints = [0.005, 0.7] as [NSNumber]
+    
+    /// The user should be able to access thisdata as well as the size of the data
+    public var heatMapPoints = [GMUWeightedLatLng]()
+    
+    // MARK: Functions that parse given data needed to build an interpolated heat map from
+    
+    /// Takes in the data set file and calculates the weights of the given points; these values are used for interpolation later
+    ///
+    /// - Parameter file: The input data set file; it must have a json extension.
+    public func setData(file: String) {
+        data.removeAll()
+        do {
+            guard let path = Bundle.main.url(forResource: file, withExtension: "json") else {
+                print("Data set path error")
+                return
+            }
+            let data = try Data(contentsOf: path)
+            let json = try JSONSerialization.jsonObject(with: data, options: [])
+            guard let object = json as? [[String: Any]] else {
+                print("Could not read the JSON file or file is empty")
+                return
+            }
+            for item in object {
+                let lat: Double = item["lat"] as? CLLocationDegrees ?? 0.0
+                let lng: Double = item["lng"] as? CLLocationDegrees ?? 0.0
+                append(lat: lat, long: lng)
+            }
+        } catch {
+            print(error.localizedDescription)
+        }
+    }
+    
+    public func addPoints(pointList: [[Double]]) {
```

Comment:
This method needs documentation. Also, I think `[CLLocationCoordinate2D]` is a better input here so that you don't need to do bounds checking for each point.

Reply:
Will do; I'll also add that to comment to make that clear.